platform/openide.util/test/unit/src/org/openide/util/TaskTest.java

```diff
@@ -21,25 +21,184 @@
 
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.Test;
 import org.netbeans.junit.Log;
-import org.netbeans.junit.NbTestCase;
-import org.openide.util.Exceptions;
-import org.openide.util.Task;
+import static java.lang.System.currentTimeMillis;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import org.junit.Ignore;
 
-public class TaskTest extends NbTestCase {
-    private Logger LOG;
 
-    public TaskTest(String testName) {
-        super(testName);
+public class TaskTest {
+    private static final Logger LOG = Logger.getLogger("org.openide.util.TaskTest");
+
+    private volatile boolean runHasBeenExecuted = false;
+    private volatile Task executedListenerTask = null;
+
+    //--------------------------------------------------------------------------
+    private static void assertFinished(final Task task) {
+
+        assertTrue(task.isFinished());
+    }
+
+    //--------------------------------------------------------------------------
+    private static void assertWaitFinishedReturnsImmediately(final Task task) {
+
+        final long begin = currentTimeMillis();
+        task.waitFinished();
+        final long duration = currentTimeMillis() - begin;
+        // shorter than 1 milisecond
+        assertEquals("The Task.waitFinished() took longer than milisecond. "
+                + "This is not neseserily a bug.", 0, duration);
```

Comment:
at the very least:
 * use `nanoTime`
 * allow a delta of 1-2 ms

Most tests should have enough heap to hopefully not trigger GC in situations like this (which would cause failure too).

Reply:
I change currentTimeMillis() to nanoTime() and expanded wait time to 10 miliseconds in both tests.
10 miliseconds steel feel "immediately" from users perspective and the false negative rate should be really low.
now these two tests, though imperfect guard us from "catastrophic" failures (like infinite wait); alternatively we can skip testing these use cases entirely;

if You accept, I'll squash it.

Reply:
thanks. Looks good. 

Infinite wait is somewhat covered by test and job timeouts - so things can't get stuck forever.

When squashing please make sure to only leave the interesting parts in the commit message to describe what the changes do, so that we have less noise in the log. No need to mention all the individual review motivated steps in the squashed commit msg.